### PR TITLE
Fix detect_os errors

### LINKED ISSUE #137: don't use detect_os.js on the downloads page
It's only designed for the homepage

### LINKED ISSUE #138: embed detect_os.js in the binary and emit it
Currently we just expect to find it at src/site/javascript/detect_os.js

### PR DESCRIPTION
**Pull request by SaraVieira**
- Removes thes script from everywhere but the homepage
- Removes the read from file and uses the JS a string that can be read by rust

closes https://github.com/axodotdev/oranda/issues/138
closes https://github.com/axodotdev/oranda/issues/137

### CODE REVIEW

File: src/site/artifacts.rs [hunk 1]
```diff
@@ -6,10 +6,9 @@ use axohtml::elements::{div, script, span};
 use axohtml::{html, text, unsafe_text};
 use cargo_dist_schema::{Artifact, ArtifactKind, DistManifest};
 
+use crate::site::javascript;
```
Comment (ashleygwilliams) [inline src/site/artifacts.rs]: i was so close to creating this in #135 - i dig it!!

File: src/site/artifacts.rs [hunk 2]
```diff
@@ -6,10 +6,9 @@ use axohtml::elements::{div, script, span};
...
 use crate::site::markdown::syntax_highlight::syntax_highlight;
 use crate::site::markdown::syntax_highlight::syntax_themes::SyntaxTheme;
-use std::fs::File;
```
Comment (ashleygwilliams) [inline src/site/artifacts.rs:11]: chef's kiss

File: src/site/artifacts.rs [hunk 3]
```diff
@@ -122,12 +121,14 @@ pub fn get_install_hint(
     String::new()
 }
 
-pub fn get_os_script(config: &Config) -> Result<Box<script<String>>> {
-    let detect_os_js = axoasset::copy("src/site/javascript/detect_os.js", &config.dist_dir);
+pub fn get_os_script(dist_dir: &String) -> Result<Box<script<String>>> {
```
Comment (ashleygwilliams) [inline src/site/artifacts.rs]: maybe move this to the new js module?

**Review by ashleygwilliams — APPROVED** [review]
some comments, but in general seems good!

File: src/site/artifacts.rs (hunk 3 above)
Comment (SaraVieira) [reply src/site/artifacts.rs]: Done!

**Pull request merged**